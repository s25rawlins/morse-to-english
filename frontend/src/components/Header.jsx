import React from 'react';

const Header = () => {
  return (
    <header className="header">
      <h1>Morse Code Translator</h1>
      <p>
        Convert between English text and Morse code with ease. 
        Our intelligent translator handles ambiguous patterns and 
        provides multiple interpretations when needed.
      </p>
    </header>
  );
};

export default Header;
