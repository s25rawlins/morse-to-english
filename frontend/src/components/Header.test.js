import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Header from './Header';

describe('Header Component', () => {
  test('renders header with correct title', () => {
    render(<Header />);
    
    expect(screen.getByText('Morse Code Translator')).toBeInTheDocument();
  });

  test('has correct semantic structure', () => {
    render(<Header />);
    
    const header = screen.getByRole('banner');
    expect(header).toBeInTheDocument();
    expect(header).toHaveClass('header');
  });

  test('contains proper heading level', () => {
    render(<Header />);
    
    const heading = screen.getByRole('heading', { level: 1 });
    expect(heading).toBeInTheDocument();
    expect(heading).toHaveTextContent('Morse Code Translator');
  });


  test('renders consistently', () => {
    const { rerender } = render(<Header />);
    
    expect(screen.getByText('Morse Code Translator')).toBeInTheDocument();
    
    rerender(<Header />);
    
    expect(screen.getByText('Morse Code Translator')).toBeInTheDocument();
  });

  test('accessibility attributes', () => {
    render(<Header />);
    
    const header = screen.getByRole('banner');
    expect(header).toBeInTheDocument();
    
    const heading = screen.getByRole('heading', { level: 1 });
    expect(heading).toBeInTheDocument();
  });

  test('snapshot consistency', () => {
    const { container } = render(<Header />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
