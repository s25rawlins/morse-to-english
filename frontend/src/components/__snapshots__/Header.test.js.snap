// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component snapshot consistency 1`] = `
<header
  class="header"
>
  <h1>
    Morse Code Translator
  </h1>
  <p>
    Convert between English text and Morse code with ease. Our intelligent translator handles ambiguous patterns and provides multiple interpretations when needed.
  </p>
</header>
`;
